src/google/protobuf/stubs/platform_macros.h

```diff
@@ -115,6 +115,8 @@ GOOGLE_PROTOBUF_PLATFORM_ERROR
 // TLS support on android.
 // iOS also does not support the __thread keyword.
 #define GOOGLE_PROTOBUF_NO_THREADLOCAL
+
+#include <pthread.h>
```

Comment:
For iOS and Android, GOOGLE_PROTOBUF_NO_THREADLOCAL will be defined, and as basically every source file includes platform_macros.h (directly or transitively), every source file will include <pthread.h> for iOS and Android build. This doesn't seem very nice to me.

How many files are using GOOGLE_PROTOBUF_NO_THREADLOCAL? If it's just 2 or 3, I think it's better to duplicate "include <pthread.h>" in those files.